### PR TITLE
Add a separate job to prepare managed node

### PR DESCRIPTION
**Pull request by spetrosi**
Managed node preparation requires more complicated structure than oneliners in FMF plan, this PR creates a separate FMF test to be run on managed nodes to prepare for tests.

### CODE REVIEW

**Comment by spetrosi** [comment]
@richm this is ready for review.
I reworked disk_provisioner.sh into a plain shell and simplified it a bit.
This requires changes to test plans, so I'll need to update plans in all roles. Note that plans/general.fmf in this repo is just a illustration, it is not used anywhere, roles use their local plans.

File: library/lib.sh [hunk 1]
```diff
@@ -234,8 +234,13 @@ rolesRunPlaybook() {
     local test_playbook=$2
     local inventory=$3
     local skip_tags=$4
-    LOGFILE="${test_playbook%.*}"-ANSIBLE-"$ANSIBLE_VER"
-    rlRun "ansible-playbook -i $inventory $skip_tags $tests_path$test_playbook -v &> $LOGFILE" 0 "Test $test_playbook with ANSIBLE-$ANSIBLE_VER"
+    local LOGFILE="${test_playbook%.*}"-ANSIBLE-"$ANSIBLE_VER"
+    # If LSR_DEBUG is true, print output to terminal
+    if [ "$LSR_DEBUG" == true ] || [ "$LSR_DEBUG" == True ]; then
+        rlRun "DEFAULT_LOG_PATH=$LOGFILE ansible-playbook -i $inventory $skip_tags $tests_path$test_playbook -v" 0 "Test $test_playbook with ANSIBLE-$ANSIBLE_VER"
```
Comment (richm) [inline library/lib.sh:240]: I think we should use `-vv` in order to get exact file/line for tasks.

File: library/lib.sh [hunk 2]
```diff
@@ -234,8 +234,13 @@ rolesRunPlaybook() {
...
+    else
+        rlRun "ansible-playbook -i $inventory $skip_tags $tests_path$test_playbook -v &> $LOGFILE" 0 "Test $test_playbook with ANSIBLE-$ANSIBLE_VER"
```
Comment (richm) [inline library/lib.sh:242]: here too

File: library/lib.sh [hunk 3]
```diff
@@ -250,6 +255,104 @@ rolesRunPlaybook() {
...
+    if ! grep -q drive: "$provisionfmf"; then
+        rlRun "rm -rf ${role_path}"
+        return
+    fi
+    if type -p python3; then
+        PYTHON=python3
+    elif type -p python; then
+        PYTHON=python
+    elif type -p python2; then
+        PYTHON=python2
```
Comment (richm) [inline library/lib.sh:317]:
```bash
    elif  [ -f /usr/libexec/platform-python ]; then
        PYTHON=/usr/libexec/platform-python
```

File: library/lib.sh [hunk 4]
```diff
@@ -250,6 +255,104 @@ rolesRunPlaybook() {
...
+        rlRun "$PYTHON -m pip install yq"
+    fi
+    if ! type -p targetcli; then
+        rlRun "yum install targetcli -y"
+    fi
+    disks=$(yq '."standard-inventory-qcow2".qemu.drive[].size' "$provisionfmf")
+    # Nothing to do
+    [ -z "$disks" ] && return
+
+    disk_provisioner_dir=$(mktemp --directory)
```
Comment (richm) [inline library/lib.sh:333]: ok - for now - we'll see if we run into problems with not enough disk space in /tmp

File: library/lib.sh (hunk 3 above)
Comment (spetrosi) [reply library/lib.sh:317]: What's this for?

Comment (richm) [reply library/lib.sh:317]: It is another place to look for a python interpreter - most RHEL have this as the "default" python interpreter.

File: library/lib.sh (hunk 4 above)
Comment (spetrosi) [reply library/lib.sh]: Modified like you suggested

**Review by richm — APPROVED** [review]

**Pull request merged**